core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/ExcelInputReader.java

```diff
@@ -0,0 +1,457 @@
+
+    getOutputSchema();
+    StructuredRecord.Builder builder = StructuredRecord.builder(outputSchema);
+    String inputValue = input.getValue().toString();
+
+    String[] excelRecords = inputValue.split("\t");
+
+    String fileName = excelRecords[1];
+    String sheetName = excelRecords[2];
+    String ifEndRow = excelRecords[3];
+
+    int prevRowNum = Integer.parseInt(excelRecords[0]);
+
+    if (filePrevRowNumMap.containsKey(fileName)) {
+      if (prevRowNum - filePrevRowNumMap.get(fileName) > 1 &&
+        excelInputreaderConfig.terminateIfEmptyRow.equalsIgnoreCase("true")) {
+        throw new ExecutionException("Encountered empty row while reading Excel file :" + fileName +
+                                       " . Terminating processing", new Throwable());
+      }
+    }
+
+    filePrevRowNumMap.put(fileName, prevRowNum);
+
+    Map<String, String> excelColumnValueMap = new HashMap<>();
+
+    for (String columns : excelRecords) {
+      String[] columnValueArray = columns.split("\r");
+      if (columnValueArray.length > 1) {
+        String columnName = columnValueArray[0];
+        String columnValue = columnValueArray[1];
+
+        if (columnMapping.containsKey(columnName)) {
+          excelColumnValueMap.put(columnMapping.get(columnName), columnValue);
+        } else {
+          excelColumnValueMap.put(columnName, columnValue);
+        }
+      }
+    }
+
+    try {
+      for (Schema.Field field : outputSchema.getFields()) {
+        String fieldName = field.getName();
+        if (excelColumnValueMap.containsKey(fieldName)) {
+          builder.convertAndSet(fieldName, excelColumnValueMap.get(fieldName));
+        } else {
+          builder.set(fieldName, NULL);
+        }
+      }
+
+      builder.set(FILE, new Path(fileName).getName());
+      builder.set(SHEET, sheetName);
+
+      if (ifEndRow.equalsIgnoreCase(END)) {
+        trackProcessedFiles(fileName);
+      }
+
+      emitter.emit(builder.build());
+    } catch (Exception e) {
+      switch (excelInputreaderConfig.ifErrorRecord) {
+        case EXIT_ON_ERROR:
+          throw new IllegalStateException("Terminating processing on error : " + e.getMessage());
+        case WRITE_ERROR_DATASET:
+          StructuredRecord.Builder errorRecordBuilder = StructuredRecord.builder(errorRecordSchema);
+          errorRecordBuilder.set(KEY, fileName + "_" + sheetName + "_" + excelRecords[0]);
+          errorRecordBuilder.set(FILE, fileName);
+          errorRecordBuilder.set(SHEET, sheetName);
+          errorRecordBuilder.set(RECORD, inputValue);
+          writeErrorRecordToDataset(errorRecordBuilder.build());
+          break;
+        default:
+          //ignore on error
+          break;
+      }
+
+    }
+
+  }
+
+  /**
+   * Writes proceesed file name to memory table.
+   * @param fileName
+   */
+  private void trackProcessedFiles(String fileName) {
```

Comment:
Why do we need this two line function and it's used in only one place. Makes it more readable.